### PR TITLE
allow closing via window manager

### PR DESCRIPTION
**Pull request by erpo41**

### CODE REVIEW

File: src/application.cpp [hunk 1]
```diff
@@ -104,6 +104,9 @@ void Application::run() {
       if(event.type() == SDL_KEYDOWN) {
         current_state()->key_down(event);
       }
+      if(event.type() == SDL_QUIT) {
```
Comment (williamkeller) [inline src/application.cpp]: Let's move this above the check for keydown, and add a `break;` statement after the stop_running() call. That will basically shortcut any other events that might be pending and immediately exit.

**Comment by williamkeller** [comment]
I added a comment on the PR. Let me know if you get an automatic reply for
it.

File: src/application.cpp [hunk 2]
```diff
@@ -104,6 +104,9 @@ void Application::run() {
...
+	app.stop_running();
```
Comment (williamkeller) [inline src/application.cpp]: You don't need `app.` here since it's already inside the Application class.

**Comment by williamkeller** [comment]
Doh! Missing closing brace on the if block! Kicking myself for not catching that before I merged.

**Comment by erpo41** [comment]
I did get the comments and briefly attempted to fix a merge conflict before
realizing that the new else if structure in application.cpp in the player
branch obviates the need for a break; in the xclose branch. You might have
some pull request notifications for xclose updates, but I've already
deleted them.

Things are looking good. So far I'm into chapter 4 of SDL game development
and it's exciting!

**Comment by williamkeller** [comment]
Which book are you reading? I think you've mentioned the title before, but
I can't find it.

We're getting close to actually working on gameplay! We could do collision
detection next, then load a map level, and be able to wander around.

**Comment by erpo41** [comment]
It's SDL Game Development by Shaun Mitchell. I haven't read the section on collision detection yet, but that sounds pretty reasonable.

On a related note, I put my new SDL skills to work today building a simple frontend for our security camera systems. I needed a fullscreen UI to display video feeds from a couple of different DVRs on a TV, and SDL (via pygame) seemed the natural choice. Fun!

**Pull request merged**